astropy/modeling/tests/test_core.py

```diff
@@ -481,6 +486,26 @@ def test_rename_2d(model_class):
     assert new_model.name == 'Test2D'
 
 
+def test_fix_inputs_integer():
+    m = models.Identity(2)
+
+    mf = models.fix_inputs(m, {1: 22})
+    assert mf(1) == (1, 22)
+
+    mf_int32 = models.fix_inputs(m, {np.int32(1): 33})
+    assert mf_int32(1) == (1, 33)
+
+    mf_int64 = models.fix_inputs(m, {np.int64(1): 44})
+    assert mf_int64(1) == (1, 44)
```

Comment:
There's an existing test

```
astropy/modeling/tests/test_compound.py::test_fix_inputs_invalid
```

It tests that certain invalid inputs raise the appropriate errors.  You may want to add a non-integer case to it to verify it is working as intended.

Reply:
I extended `astropy/modeling/tests/test_compound.py::test_fix_inputs_invalid` to include tests that involve numpy integer keys.

Should the two `fix_inputs` tests I added to `test_core.py` be moved to `test_compound.py` so that similar tests are all together?

Reply:
Yeah, those aren't specifically compound model tests.  If anything, `test_fix_inputs_invalid` could be moved to test_core.py, but not super important.  Feel free to leave them where they are and leave cleanup to a future PR.

In terms of the diff in the PR, it's always easier to review if changes to the tests are separate from moving unchanged tests around.  👍